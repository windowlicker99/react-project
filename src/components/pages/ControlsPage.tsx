import React, { FC, useState } from 'react';
import { Tab, TabPanel, TabList, Tabs } from 'react-tabs';
import { useSelector } from 'react-redux';
import { PageSearchbar } from '@components/page-contents/PageSearchbar';
import { Table } from '@components/page-contents/Table';
import { Footer } from '@components/modules/Footer';
import classNames from 'classnames';
import { Preloader } from '@components/common/Preloader';
import { Btn } from '@components/common/form-controls/Btn';
import { selectColumnsWithDropdownList } from '@/store/selectors/columns';
import { controlsTabs } from '@/constants/tabsLists';

export const ControlsPage: FC = () => {
  const [searchText, setSearchText] = useState<string>('');

  const [loading, setLoading] = useState(true);

  const customizedColumns = useSelector(selectColumnsWithDropdownList);

  const [selectedFindingsTabIndex, setSelectedFindingsTabIndex] = useState(0);

  return (
    <Tabs
      className="content"
      id="controls"
      onSelect={setSelectedFindingsTabIndex}
      selectedIndex={selectedFindingsTabIndex}
      selectedTabPanelClassName="box-table box-not-radius"
    >
      <div className="box page-controls">
        <PageSearchbar
          searchText={searchText}
          onSearch={setSearchText}
          onCustomizeColumnsClick={(): null => null} // temp
          btnText="Create Control"
          btnIcon="icon-plus"
          showFilterBtn={false}
        />
      </div>

      <TabList className="tabs-box">
        {controlsTabs.map((tab, i) => (
          <Tab key={tab.name}>
            <Btn
              btnStatus={classNames('btn-stretch', selectedFindingsTabIndex === i ? 'btn-lg-primary' : 'btn-lg-info')}
              btnText={tab.name}
            />
          </Tab>
        ))}
      </TabList>

      {controlsTabs.map((tab) => (
        <TabPanel key={tab.name}>
          <Table
            changeLoadingState={setLoading}
            columns={customizedColumns}
            setSearchText={setSearchText}
            searchText={searchText}
            tableType={tab.type}
            withoutPagination
          />
        </TabPanel>
      ))}
      {loading ? <Preloader /> : null}
      <Footer />
    </Tabs>
  );
};
