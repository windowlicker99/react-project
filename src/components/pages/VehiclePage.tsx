import React, { FC, useState } from 'react';
import { PageControls } from '@components/page-contents/PageControls';
import { VehicleContent } from '@components/page-contents/VehicleContent';
import { CreateVehicle } from '@components/page-contents/CreateVehicle';
import { Preloader } from '@components/common/Preloader';

export const VehiclePage: FC = () => {
  const [isOpenVehicleModal, setIsOpenVehicleModal] = useState<boolean>(false);

  const [loading, setLoading] = useState(true);

  const onOpenVehicleModalClick = () => {
    setIsOpenVehicleModal(true);
  };

  const onCloseVehicleModalClick = () => {
    setIsOpenVehicleModal(false);
  };

  return (
    <div className="content" id="vehicle">
      <PageControls buttonText="Create Vehicle" onButtonClick={onOpenVehicleModalClick} />
      <VehicleContent changeLoadingState={setLoading} updateContent={isOpenVehicleModal} />
      <CreateVehicle isOpen={isOpenVehicleModal} onClose={onCloseVehicleModalClick} />
      {loading ? <Preloader /> : null}
    </div>
  );
};
