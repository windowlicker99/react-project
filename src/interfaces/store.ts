import { ETableTypes, IColumn, ICurrentElement, IElementsTabsColumnsList, IUser } from '@interfaces/interfaces';

export interface IAuth {
  user: IUser;
}
export interface IPagination {
  nextPage: () => void;
  prevPage: () => void;
  pickPage: (indexPage: number) => void;
  disablePrev: boolean;
  disableNext: boolean;
  currentPage: number;
  pageAmount: number;
  currentTab: string;
  totalElement: number;
  totalElementOnPage: number;
  showedElements: number;
}

export interface IVehicleStore {
  currentVehicle: string;
}

export interface IHeaderStore {
  subtitles: string[];
}

export interface IChangeTypeAction {
  type: ETableTypes;
}

export interface IUpdateDisplayedColumnsAction {
  displayColumnsList: IDisplayColumnsList;
}

export interface ICustomizeColumnState {
  type: ETableTypes;
  columns: IElementsTabsColumnsList;
}

export interface IStoreState {
  auth: IAuth;
  pagination: IPagination;
  vehicles: IVehicleStore;
  customizeColumns: ICustomizeColumnState;
  header: IHeaderStore;
  element: IElementStore;
}

export interface IDisplayColumnsList {
  [key: string]: boolean;
}

export interface ICustomizeColumnAction {
  type: ETableTypes;
  displayColumnsList: IDisplayColumnsList;
}

export interface ICustomizeColumnsState {
  [key: string]: IColumn[];
}

export interface IElementStore {
  currentElement: ICurrentElement;
}
