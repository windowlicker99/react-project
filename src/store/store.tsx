import { combineReducers } from 'redux';
import { configureStore } from '@reduxjs/toolkit';
import storage from 'redux-persist/lib/storage';
import { persistStore, persistReducer } from 'redux-persist';
import { authReducer } from '@store/slices/authSlice';
import { columnsReducer } from '@store/slices/columnsSlice';
import { paginationReducer } from '@store/slices/paginationSlice';
import { vehiclesReducer } from '@store/slices/vehicleSlice';
import { headerReducer } from '@store/slices/headerSlice';
import { elementReducer } from '@store/slices/elementSlice';

const reducers = combineReducers({
  auth: authReducer,
  vehicles: vehiclesReducer,
  customizeColumns: columnsReducer,
  pagination: paginationReducer,
  header: headerReducer,
  element: elementReducer,
});

const persistConfig = {
  key: 'root',
  storage,
  whiteList: ['auth', 'customizeColumns'],
};

const persistedReducer = persistReducer(persistConfig, reducers);

export const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) => [
    ...getDefaultMiddleware({
      serializableCheck: false,
    }),
  ],
});

export const persistor = persistStore(store);
