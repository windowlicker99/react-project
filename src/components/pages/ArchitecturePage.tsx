import React, { useState, FC, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { ArchitectureControls } from '@components/page-contents/ArchitectureControls';
import { Btn } from '@components/common/form-controls/Btn';
import { ArchitectureScheme } from '@components/page-contents/ArchitectureScheme';
import { ModalCenter } from '@components/modules/ModalCenter';
import { AssignRA } from '@components/page-contents/AssignRA';
import { ArchitectureAside } from '@components/common/ArchitectureAside';
import { ArchitectureDetailsContent } from '@components/common/aside-content/ArchitectureDetailsContent';
import { EAside } from '@interfaces/enums';
import { selectCurrentVehicle } from '@store/selectors/vehicle';
import { getCurrentVehicle } from '@utils/getCurrentVehicle';
import { IGatewayResponse, IVehicle } from '@interfaces/interfaces';
import { DomainDetails } from '@components/page-contents/DomainDetails';
import { ECUDetails } from '@components/page-contents/ECUDetails';
import { getVehicleGateway } from '@utils/getVehicleGateway';
import { setCurrentElement } from '@store/slices/elementSlice';
import { InitiateRiskAssesment } from '@components/page-contents/InitiateRiskAssesment';
import { currentElementSelector } from '@store/selectors/element';
import { getSchemeData } from '@utils/getSchemeData';
import { IArchitectureDomain } from '@interfaces/ArchitectureScheme/interfaces';
import { Preloader } from '@components/common/Preloader';
import { setHeader } from '@/store/slices/headerSlice';

export const ArchitecturePage: FC = () => {
  const dispatch = useDispatch();
  const [vehicle, setVehicle] = useState<IVehicle>(null);
  const [schemeData, setSchemeData] = useState<IArchitectureDomain[]>(null);
  const [isOpenDomainModal, setIsOpenDomainModal] = useState<boolean>(false);
  const [isOpenECUModal, setIsOpenECUModal] = useState<boolean>(false);
  const [isOpenHistoryView, setIsOpenHistoryView] = useState<boolean>(false);
  const [isOpenRAModal, setIsOpenRAModal] = useState<boolean>(false);
  const [isOpenInitiateView, setIsOpenInitiateView] = useState<boolean>(false);
  const [gateway, setGateway] = useState<IGatewayResponse>(null);
  const [loading, setLoading] = useState(true);

  const currentElement = useSelector(currentElementSelector);

  const currentVehicle = useSelector(selectCurrentVehicle);

  const onOpenDomainModalClick = () => {
    setIsOpenDomainModal(true);
  };

  const onCloseDomainModalClick = () => {
    setIsOpenDomainModal(false);
  };

  const onOpenECUModalClick = () => {
    setIsOpenECUModal(true);
  };

  const onCloseECUModalClick = () => {
    setIsOpenECUModal(false);
  };

  const onOpenHistoryView = () => {
    setIsOpenHistoryView(true);
  };

  const onCloseHistoryView = () => {
    setIsOpenHistoryView(false);
  };

  const onOpenInitiateView = () => {
    setIsOpenInitiateView(true);
  };

  const onOpenRAModal = () => {
    setIsOpenRAModal(true);
  };
  const onCloseRAModal = () => {
    setIsOpenRAModal(false);
  };

  const onApplyModal = () => {
    console.log('apply'); // temp
  };

  const getVehicle = async () => {
    if (!currentVehicle) {
      setVehicle(null);
      return;
    }
    const currentVehicleData = await getCurrentVehicle(currentVehicle);
    setVehicle(currentVehicleData);
  };

  const onCloseInitiateView = () => {
    setIsOpenInitiateView(false);
    getVehicle();
  };

  const getController = async (vehicleId: string) => {
    try {
      setLoading(true);
      const { gateway, type } = await getVehicleGateway(vehicleId);
      setGateway({ gateway, type });
      dispatch(setCurrentElement({ type: EAside[type], id: gateway?.id }));
      setLoading(false);
    } catch (error) {
      dispatch(setCurrentElement(null));
    }
  };

  const getArchitectureSchemeData = async (id: string) => {
    const newSchemeData = await getSchemeData(id);
    setSchemeData(newSchemeData);
    setLoading(false);
  };

  useEffect(() => {
    if (!isOpenDomainModal || isOpenECUModal) {
      getArchitectureSchemeData(currentVehicle);
    }
  }, [currentVehicle, isOpenDomainModal, isOpenECUModal]);

  useEffect(() => {
    getVehicle();
  }, [currentVehicle]);

  useEffect(() => {
    if (currentVehicle && !gateway && !isOpenECUModal) {
      getController(currentVehicle);
    }
  }, [currentVehicle, isOpenECUModal]);

  useEffect(() => {
    if (vehicle) {
      const subtitles = [];
      if (vehicle.model) {
        subtitles.push(vehicle.model);
      }

      if (vehicle.brand?.name) {
        subtitles.push(vehicle?.brand?.name);
      }
      dispatch(setHeader(subtitles));
    }
  }, [vehicle]);

  return (
    <div className="content" id="architecture">
      <ArchitectureControls
        vehicle={vehicle}
        onOpenDomainModalClick={onOpenDomainModalClick}
        onOpenHistoryView={onOpenHistoryView}
      />
      <div className="page-content">
        {loading ? (
          <Preloader />
        ) : (
          <>
            <div className="page-thumb">
              <div className="architecture-scheme" id="architectureScheme">
                <ArchitectureScheme
                  gatewayInfo={gateway}
                  schemeData={schemeData}
                  onECUCreateClick={onOpenECUModalClick}
                />
              </div>
              <div className="architecture-controls">
                <div className="page-controls-main">
                  <div className="btn-box">
                    <Btn btnStatus="btn-control" btnIcon="icon-grid" />
                    <Btn btnStatus="btn-control" btnIcon="icon-list" />
                  </div>
                  <div className="btn-box">
                    <Btn btnStatus="btn-control" btnIcon="icon-maximize" />
                    <div className="btn-box-separator" />
                    <Btn btnStatus="btn-control" btnIcon="icon-minus" />
                  </div>
                </div>
                <Btn btnStatus="btn-control" btnIcon="icon-chevrons-right" />
              </div>
            </div>
            <ArchitectureAside currentElement={currentElement} asideStatus="large">
              <ArchitectureDetailsContent
                onAssign={onOpenRAModal}
                element={currentElement}
                isAssignModalOpen={isOpenRAModal}
                onOpenInitiateView={onOpenInitiateView}
                isInitiateModalOpen={isOpenInitiateView}
              />
            </ArchitectureAside>
          </>
        )}
      </div>
      <DomainDetails
        vehicle={vehicle}
        isOpenDomainModal={isOpenDomainModal}
        onCloseDomainModalClick={onCloseDomainModalClick}
      />
      <ECUDetails vehicle={vehicle} isOpen={isOpenECUModal} onClose={onCloseECUModalClick} />
      <AssignRA element={currentElement} isOpen={isOpenRAModal} onClose={onCloseRAModal} />
      {/* Add content to modals */}
      <ModalCenter title="History" isOpen={isOpenHistoryView} onApply={onApplyModal} onClose={onCloseHistoryView} />
      <InitiateRiskAssesment
        element={currentElement}
        vehicle={vehicle}
        isOpenInitiateView={isOpenInitiateView}
        onCloseInitiateView={onCloseInitiateView}
      />
    </div>
  );
};
