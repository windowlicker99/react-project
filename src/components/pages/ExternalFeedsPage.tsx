import React, { FC, useState, useEffect } from 'react';
import { useSelector } from 'react-redux';
import { Tab, TabList, TabPanel, Tabs } from 'react-tabs';
import { Heading } from '@components/common/Heading';
import { Table } from '@components/page-contents/Table';
import { Aside } from '@components/common/Aside';
import { CardAlert } from '@components/common/CardAlert';
import { ETableTypes, IAlert } from '@interfaces/interfaces';
import { selectColumnsWithExpandAllDropdownList } from '@store/selectors/columns';
import { getData } from '@utils/getElements';
import { API_PATHS } from '@constants/paths';
import { Preloader } from '@components/common/Preloader';

export const ExternalFeedsPage: FC = () => {
  const [alerts, setAlerts] = useState<IAlert[]>([]);
  const [loading, setLoading] = useState(true);

  const columns = useSelector(selectColumnsWithExpandAllDropdownList);

  const getAlerts = async () => {
    const newAlerts: IAlert[] = await getData(API_PATHS.alerts);
    setAlerts(newAlerts);
  };

  useEffect(() => {
    getAlerts();
  }, []);

  return (
    <Tabs className="page-holder" id="feeds">
      <div className="aside-holder">
        {!loading ? <Heading headingText="Car Alerts" headingNumber={`(${alerts.length})`} /> : null}
        <Aside asideStatus="small transparent">
          <TabList>
            {alerts.map((alert) => (
              <Tab className="tab-item" selectedClassName="active" key={`tab-${alert.name}`}>
                <CardAlert alert={alert} />
              </Tab>
            ))}
          </TabList>
        </Aside>
      </div>

      {alerts.map((alert) => (
        <TabPanel className="tab-content" key={`tabpanel-${alert.name}`}>
          <Heading headingText="Vehicle Types" headingNumber={`(${alert?.vehicleIds.length})`} subtitle={alert.name} />
          <div className="box-table">
            <Table
              changeLoadingState={setLoading}
              tableType={ETableTypes.feeds}
              columns={columns}
              params={{ alertId: alert.id }}
              showHeader
            />
          </div>
        </TabPanel>
      ))}
      {loading ? <Preloader /> : null}
    </Tabs>
  );
};
