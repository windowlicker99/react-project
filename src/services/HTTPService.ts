import axios, { AxiosPromise } from 'axios';
import { IParams } from '@interfaces/interfaces';

const baseUrl = (path: string) => `${process.env.API_URL || ''}${path}`;

// eslint-disable-next-line @typescript-eslint/no-explicit-any
type TData = Record<string, any>;
export default class HTTPService {
  static get(path = '', params: IParams = {}) {
    return axios({
      method: 'GET',
      url: baseUrl(path),
      params,
    });
  }

  static post(path: string, data: TData): AxiosPromise {
    return axios({
      method: 'POST',
      url: baseUrl(path),
      data,
      headers: {
        'Content-Type': 'application/json',
      },
    });
  }

  static patch(path: string, data: TData): AxiosPromise {
    return axios({
      method: 'PATCH',
      url: baseUrl(path),
      data,
    });
  }

  static put(path: string, data: TData): AxiosPromise {
    return axios({
      method: 'PUT',
      url: baseUrl(path),
      data,
    });
  }

  static delete(path: string): AxiosPromise {
    return axios({
      method: 'DELETE',
      url: baseUrl(path),
    });
  }
}
