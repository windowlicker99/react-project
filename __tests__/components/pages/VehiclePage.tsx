import React from 'react';
import { render, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import { store } from '@store/store';
import { VehiclePage } from '@components/pages/VehiclePage';
import HTTPService from '@services/HTTPService';

describe('VehiclePage tests', () => {
  const brands = [
    {
      id: '0',
      name: 'BMW',
    },
    {
      id: '1',
      name: 'Cadillac',
    },
  ];

  it('Should render', async () => {
    const mock = new MockAdapter(axios);
    mock.onGet('/brands').reply(200, brands);

    jest.spyOn(HTTPService, 'get').mockResolvedValue(axios.get('/brands'));

    global.URL.revokeObjectURL = jest.fn();
    const { container } = render(
      <Provider store={store}>
        <VehiclePage />
      </Provider>
    );
    await waitFor(() => expect(container.getElementsByClassName('content').length).toEqual(1));
  });

  it('Should match snapshot', async () => {
    const mock = new MockAdapter(axios);
    mock.onGet('/brands').reply(200, brands);

    jest.spyOn(HTTPService, 'get').mockResolvedValue(axios.get('/brands'));

    global.URL.revokeObjectURL = jest.fn();
    const { asFragment } = render(
      <Provider store={store}>
        <VehiclePage />
      </Provider>
    );
    await waitFor(() => expect(asFragment()).toMatchSnapshot());
  });
});
