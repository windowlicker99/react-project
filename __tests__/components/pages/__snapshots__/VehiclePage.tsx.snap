// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehiclePage tests Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="content"
    id="vehicle"
  >
    <div
      class="box page-controls"
    >
      <div
        class="page-controls-content flex-end"
      >
        <div
          class="page-controls-item"
        >
          <div
            class="mr-15"
          >
            <div
              class="form-group"
            >
              <label
                class="form-label search-label"
                for="exampleFormControlInput1"
              >
                What are you looking for?
              </label>
              <div
                class="search-group"
              >
                <div
                  class="search-icon"
                >
                  <span
                    class="icon-loupe"
                  />
                </div>
                <input
                  class="form-control search-input"
                  id="exampleFormControlInput1"
                  placeholder="Search"
                  type="email"
                  value=""
                />
                <button
                  class="search-btn"
                  type="button"
                >
                  <span
                    class="icon-cross"
                  />
                </button>
              </div>
            </div>
          </div>
          <div
            class="mr-15"
          >
            <button
              class="btn-xs-primary"
              type="button"
            >
              <span
                class="icon-filter-blue"
              />
            </button>
          </div>
        </div>
        <div
          class="page-controls-item"
        >
          <button
            class="btn btn-lg-primary"
            type="button"
          >
            <span
              class="btn-icon icon-plus"
            />
            <span
              class="btn-text"
            >
              Create Vehicle
            </span>
          </button>
        </div>
      </div>
      <div
        class="page-filter"
      >
        <div
          class="page-filter-inner"
        />
        <div
          class="page-filter-btn"
        >
          <button
            class="btn btn-xs-secondary mr-15"
            type="button"
          >
            <span
              class="btn-icon icon-close"
            />
            <span
              class="btn-text"
            />
          </button>
          <button
            class="btn btn-lg-outline"
            type="button"
          >
            <span
              class="btn-icon "
            />
            <span
              class="btn-text"
            >
              Apply
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="box-table"
    >
      <table
        class="table"
        role="table"
      >
        <thead
          class="table-header"
        >
          <tr
            class="table-row"
            role="row"
          >
            <th
              class="table-col"
              colspan="1"
              role="columnheader"
            >
              <div
                class="table-header-inner"
              >
                <span
                  class="table-header-title"
                >
                  <button
                    class="btn btn-xs-primary"
                    currentitem="false"
                    data-for="expandBtn"
                    data-tip="true"
                    type="button"
                  >
                    <span
                      class="btn-icon icon-arrow-line"
                    />
                    <span
                      class="btn-text"
                    />
                  </button>
                  <div
                    class="__react_component_tooltip expandBtn place-top type-dark"
                    data-id="tooltip"
                    id="expandBtn"
                  >
                    <style
                      aria-hidden="true"
                    >
                      
  	.expandBtn {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
  	}

  	.expandBtn.place-top {
        margin-top: -10px;
    }
    .expandBtn.place-top::before {
        border-top: 8px solid transparent;
    }
    .expandBtn.place-top::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        bottom: -6px;
        left: 50%;
        margin-left: -8px;
        border-top-color: #222;
        border-top-style: solid;
        border-top-width: 6px;
    }

    .expandBtn.place-bottom {
        margin-top: 10px;
    }
    .expandBtn.place-bottom::before {
        border-bottom: 8px solid transparent;
    }
    .expandBtn.place-bottom::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        top: -6px;
        left: 50%;
        margin-left: -8px;
        border-bottom-color: #222;
        border-bottom-style: solid;
        border-bottom-width: 6px;
    }

    .expandBtn.place-left {
        margin-left: -10px;
    }
    .expandBtn.place-left::before {
        border-left: 8px solid transparent;
    }
    .expandBtn.place-left::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        right: -6px;
        top: 50%;
        margin-top: -4px;
        border-left-color: #222;
        border-left-style: solid;
        border-left-width: 6px;
    }

    .expandBtn.place-right {
        margin-left: 10px;
    }
    .expandBtn.place-right::before {
        border-right: 8px solid transparent;
    }
    .expandBtn.place-right::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        left: -6px;
        top: 50%;
        margin-top: -4px;
        border-right-color: #222;
        border-right-style: solid;
        border-right-width: 6px;
    }
  
                    </style>
                    Expand All
                  </div>
                </span>
                <div
                  class="table-filter"
                />
              </div>
            </th>
            <th
              class="table-col"
              colspan="1"
              role="columnheader"
            >
              <div
                class="table-header-inner"
              >
                <span
                  class="table-header-title"
                >
                  BRAND
                </span>
                <div
                  class="table-filter"
                >
                  <button
                    class="table-filter-btn"
                    type="button"
                  >
                    <span
                      class="icon-arrow-full-up"
                    />
                  </button>
                  <button
                    class="table-filter-btn"
                    type="button"
                  >
                    <span
                      class="icon-arrow-full-down"
                    />
                  </button>
                </div>
              </div>
            </th>
            <th
              class="table-col"
              colspan="1"
              role="columnheader"
            >
              <div
                class="table-header-inner"
              >
                <span
                  class="table-header-title"
                >
                  MODEL
                </span>
                <div
                  class="table-filter"
                >
                  <button
                    class="table-filter-btn"
                    type="button"
                  >
                    <span
                      class="icon-arrow-full-up"
                    />
                  </button>
                  <button
                    class="table-filter-btn"
                    type="button"
                  >
                    <span
                      class="icon-arrow-full-down"
                    />
                  </button>
                </div>
              </div>
            </th>
            <th
              class="table-col"
              colspan="1"
              role="columnheader"
            >
              <div
                class="table-header-inner"
              >
                <span
                  class="table-header-title"
                >
                  PLATFORM
                </span>
                <div
                  class="table-filter"
                >
                  <button
                    class="table-filter-btn"
                    type="button"
                  >
                    <span
                      class="icon-arrow-full-up"
                    />
                  </button>
                  <button
                    class="table-filter-btn"
                    type="button"
                  >
                    <span
                      class="icon-arrow-full-down"
                    />
                  </button>
                </div>
              </div>
            </th>
            <th
              class="table-col"
              colspan="1"
              role="columnheader"
            >
              <div
                class="table-header-inner"
              >
                <span
                  class="table-header-title"
                >
                  CURRENT PHASE
                </span>
                <div
                  class="table-filter"
                />
              </div>
            </th>
            <th
              class="table-col"
              colspan="1"
              role="columnheader"
            >
              <div
                class="table-header-inner"
              >
                <span
                  class="table-header-title"
                >
                  RISK LEVEL
                </span>
                <div
                  class="table-filter"
                />
              </div>
            </th>
            <th
              class="table-col"
              colspan="1"
              role="columnheader"
            >
              <div
                class="table-header-inner"
              >
                <span
                  class="table-header-title"
                >
                  COMPLIANCE LEVEL
                </span>
                <div
                  class="table-filter"
                />
              </div>
            </th>
            <th
              class="table-col"
              colspan="1"
              role="columnheader"
            >
              <div
                class="table-header-inner"
              >
                <span
                  class="table-header-title"
                >
                  RA PROGRESS
                </span>
                <div
                  class="table-filter"
                />
              </div>
            </th>
            <th
              class="table-col"
              colspan="1"
              role="columnheader"
            >
              <div
                class="table-header-inner"
              >
                <span
                  class="table-header-title"
                >
                  LAST UPDATE
                </span>
                <div
                  class="table-filter"
                />
              </div>
            </th>
            <th
              class="table-col"
              colspan="1"
              role="columnheader"
            >
              <div
                class="table-header-inner"
              >
                <span
                  class="table-header-title"
                >
                  ACTIONS
                </span>
                <div
                  class="table-filter"
                />
              </div>
            </th>
          </tr>
        </thead>
        <tbody
          class="table-body"
          role="rowgroup"
        />
      </table>
    </div>
    <div
      class="modal modal-aside"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <div
            class="modal-heading"
          >
            <div
              class="modal-header-content"
            >
              <span
                class="modal-subtitle"
              >
                Step 1/2
              </span>
              <span
                class="modal-title"
              >
                Create Vehicle
              </span>
              <span
                class="modal-subtitle"
              >
                Basic Information
              </span>
            </div>
            <button
              class="btn btn-xs-secondary"
              type="button"
            >
              <span
                class="icon-close"
              />
            </button>
          </div>
        </div>
        <div
          class="modal-body"
        >
          <form
            action="#"
          >
             
            <div
              class="vehicle-create"
            >
              <div
                class="upload upload-icon"
                tabindex="0"
              >
                <input
                  autocomplete="off"
                  multiple=""
                  style="display: none;"
                  tabindex="-1"
                  type="file"
                />
                 
                <img
                  alt=""
                  class="upload-img"
                  src=""
                />
                <div
                  class="upload-content"
                >
                  <div
                    class="upload-text"
                  >
                    Drop a Vehicle photo here, or 
                    <span
                      class="link"
                    >
                      click to browse
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="grid-box-columns"
              >
                <div
                  class="form-group"
                >
                  <label
                    class="form-label"
                    for="exampleFormControlInput1"
                  >
                    Brand
                  </label>
                  <div
                    class=" css-2b097c-container"
                  >
                    <span
                      aria-atomic="false"
                      aria-live="polite"
                      aria-relevant="additions text"
                      class="css-1f43avz-a11yText-A11yText"
                    />
                    <div
                      class=" css-1pgkwyd-control"
                    >
                      <div
                        class=" css-g1d714-ValueContainer"
                      >
                        <div
                          class=" css-oco7ot-placeholder"
                        >
                          Brand
                        </div>
                        <div
                          class="css-b8ldur-Input"
                        >
                          <div
                            class=""
                            style="display: inline-block;"
                          >
                            <input
                              aria-autocomplete="list"
                              autocapitalize="none"
                              autocomplete="off"
                              autocorrect="off"
                              id="react-select-5-input"
                              spellcheck="false"
                              style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                              tabindex="0"
                              type="text"
                              value=""
                            />
                            <div
                              style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                            />
                          </div>
                        </div>
                      </div>
                      <div
                        class=" css-1hb7zxy-IndicatorsContainer"
                      >
                        <div
                          aria-hidden="true"
                          class=" css-tlfecz-indicatorContainer"
                        >
                          <div
                            class="form-arrow icon-arrow-border-down"
                          />
                        </div>
                      </div>
                    </div>
                    <input
                      name="brand"
                      type="hidden"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="form-group"
                >
                  <label
                    class="form-label"
                  >
                    Model
                  </label>
                  <input
                    class="form-control"
                    form="[object Object]"
                    maxlength="25"
                    name="model"
                    placeholder="Model"
                    value=""
                  />
                </div>
                <div
                  class="form-group"
                >
                  <label
                    class="form-label"
                  >
                    Platform
                  </label>
                  <input
                    class="form-control"
                    form="[object Object]"
                    maxlength="25"
                    name="platform"
                    placeholder="Platform"
                    value=""
                  />
                </div>
                <div
                  class="form-group"
                >
                  <label
                    class="form-label"
                    for="exampleFormControlInput1"
                  >
                    Phase
                  </label>
                  <div
                    class=" css-2b097c-container"
                  >
                    <span
                      aria-atomic="false"
                      aria-live="polite"
                      aria-relevant="additions text"
                      class="css-1f43avz-a11yText-A11yText"
                    />
                    <div
                      class=" css-1pgkwyd-control"
                    >
                      <div
                        class=" css-g1d714-ValueContainer"
                      >
                        <div
                          class=" css-oco7ot-placeholder"
                        >
                          Phase
                        </div>
                        <div
                          class="css-b8ldur-Input"
                        >
                          <div
                            class=""
                            style="display: inline-block;"
                          >
                            <input
                              aria-autocomplete="list"
                              autocapitalize="none"
                              autocomplete="off"
                              autocorrect="off"
                              id="react-select-6-input"
                              spellcheck="false"
                              style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                              tabindex="0"
                              type="text"
                              value=""
                            />
                            <div
                              style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                            />
                          </div>
                        </div>
                      </div>
                      <div
                        class=" css-1hb7zxy-IndicatorsContainer"
                      >
                        <div
                          aria-hidden="true"
                          class=" css-tlfecz-indicatorContainer"
                        >
                          <div
                            class="form-arrow icon-arrow-border-down"
                          />
                        </div>
                      </div>
                    </div>
                    <input
                      name="phase"
                      type="hidden"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="form-group"
                >
                  <label
                    class="form-label"
                    for="exampleFormControlInput1"
                  >
                    Regulation
                  </label>
                  <div
                    class=" css-2b097c-container"
                  >
                    <span
                      aria-atomic="false"
                      aria-live="polite"
                      aria-relevant="additions text"
                      class="css-1f43avz-a11yText-A11yText"
                    />
                    <div
                      class=" css-1pgkwyd-control"
                    >
                      <div
                        class=" css-g1d714-ValueContainer"
                      >
                        <div
                          class=" css-oco7ot-placeholder"
                        >
                          Regulation
                        </div>
                        <div
                          class="css-b8ldur-Input"
                        >
                          <div
                            class=""
                            style="display: inline-block;"
                          >
                            <input
                              aria-autocomplete="list"
                              autocapitalize="none"
                              autocomplete="off"
                              autocorrect="off"
                              id="react-select-7-input"
                              spellcheck="false"
                              style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                              tabindex="0"
                              type="text"
                              value=""
                            />
                            <div
                              style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                            />
                          </div>
                        </div>
                      </div>
                      <div
                        class=" css-1hb7zxy-IndicatorsContainer"
                      >
                        <div
                          aria-hidden="true"
                          class=" css-tlfecz-indicatorContainer"
                        >
                          <div
                            class="form-arrow icon-arrow-border-down"
                          />
                        </div>
                      </div>
                    </div>
                    <input
                      name="regulation"
                      type="hidden"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="card-dropdown open"
              >
                <div
                  class="card-dropdown-header"
                >
                  <div
                    class="controller-info"
                  >
                    <span
                      class="controller-info-title"
                    >
                      Does the vehicle have Gateway?
                    </span>
                    <div
                      class="switch"
                    >
                      <input
                        class="switch-input"
                        id="haveGateway"
                        name="haveGateway"
                        type="checkbox"
                      />
                      <label
                        class="switch-label"
                        for="haveGateway"
                      >
                        <span
                          class="switch-text"
                        />
                        <span
                          class="switch-trigger"
                        />
                      </label>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </form>
        </div>
        <div
          class="modal-footer"
        >
          <div
            class="modal-footer-control"
          >
            <div
              class="modal-footer-item"
            />
            <button
              class="btn btn-lg-primary btn-stretch"
              type="button"
            >
              <span
                class="btn-icon "
              />
              <span
                class="btn-text"
              >
                Next
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="preloader"
    >
      <div
        class="preloader-circle"
      />
      <span
        class="preloader-label"
      >
        Loading data...
      </span>
    </div>
  </div>
</DocumentFragment>
`;
